kolena/_experimental/object_detection/utils.py

```diff
@@ -99,7 +99,18 @@ def compute_optimal_f1(
         _, _, y_true_by_label, y_score_by_label = _compute_sklearn_arrays_by_class(all_bbox_matches)
         for label in sorted(y_true_by_label.keys()):
             y_true, y_score = y_true_by_label[label], y_score_by_label[label]
+
+            # python 3.7.1's sklearn is different
+            if np.all(y_true == 0):
+                optimal_thresholds[label] = 0
+                continue
```

Comment:
Would prefer not to have version-specific handling in the implementation like this. Is there a version pin we can update to ensure that users of 3.7 have a `scikit-learn` package with consistent behavior?

Reply:
sklearn1.1+ requires python3.8
For python3.7, we have [sklearn1.0](https://scikit-learn.org/1.0/modules/generated/sklearn.metrics.precision_recall_curve.html#sklearn.metrics.precision_recall_curve). The code here is different, and causes some unit tests to fail. I can remove the comment on 3.7.1, as `np.all(y_true == 0) => optimal_threshold == 0` should be true for any case, not version specific.